portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,101 @@
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns data of comments*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    public class Comment { 
+        String name; 
+        String email;
+        String comment;
+        long timestamp; 
+        long id;
+  
+        public Comment(String name, String email, String comment, long timestamp, long id) {
+            this.name = name;
+            this.email = email;
+            this.comment = comment;
+            this.timestamp =timestamp;
+            this.id= id;
+        }
+    }
+
+    
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    String userChoice = request.getParameter("num");
+    int maxComments;
+    maxComments = Integer.parseInt(userChoice);
+    int i =0;
+    for (Entity entity : results.asIterable()) {
+        if (i < maxComments){
+            long id = entity.getKey().getId();
+            String name = (String) entity.getProperty("name");
+            String email = (String) entity.getProperty("email");
+            String comment = (String) entity.getProperty("comment");
+            long timestamp = (long) entity.getProperty("timestamp");
+        
+            Comment userComment = new Comment(name, email, comment,timestamp,id);
+                    comments.add(0,userComment);        
+        }
+        i++;
+    }
+    
+
+    String json = convertToJsonUsingGson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
   }
+
+
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = request.getParameter("name");
+    String email = request.getParameter("email");
+    String comment = request.getParameter("comment");
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("name", name);
+    commentEntity.setProperty("email", email);
+    commentEntity.setProperty("comment", comment);
+    commentEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+    response.sendRedirect("/contact.html");
+  }
+
+  private String convertToJsonUsingGson(ArrayList<Comment> comments) {
```

Comment:
I believe the Gson object can be reused, so it should be possible to just put `private final Gson gson = new Gson()` at the top of the DataServlet, and just use `gson.toJson` instead of creating a new instance for every request.

Reply:
done